docs/Pflichtenheft.tex

```diff
@@ -119,39 +139,59 @@ \subsubsection{Optionale Funktionalität}
 \subsection{Überwachungskonsole}
 \subsubsection{Funktionalität}
 \begin{enumerate}
-\item[FA310] Die \gls{Uberwachungskonsole} zeigt die Farbe der Flüssigkeitstanks an.
-\item[FA320] Die \gls{Uberwachungskonsole} zeigt die Zu-, und Abflussmenge der oberen Flüssigkeitstanks an.
-\item[FA330] Die \gls{Uberwachungskonsole} zeigt die Abflussmenge des unteren Flüssigkeitstanks an.
-\item[FA340] Die \gls{Uberwachungskonsole} zeigt die Drehzahl des Mischermotors an.
-\item[FA350] Alle angezeigten \glspl{Sensordatum} werden mit der eingestellten Aktualisierungsfrequenz aktualisiert.
-\item[FA360] Der Benutzer kann die Anzeige der Füllstands- und Temperaturverläufe für jeden Tank ein- und ausschalten.
-\item[FA370] Es gibt eine .jar Datei, die bei Ausführung die \gls{Uberwachungskonsole} startet.
-\item[FA380] Es k\"onnen Alarme mit ihrem jeweiligen Ausl\"oser bei der Fertigungssimulation registriert werden.
-\item[FA385] Bereits registrierte Alarme k\"onnen wieder gel\"oscht werden.
-\item[FA390] Die \"Uberwachungskonsole kann sich unter Angabe einer IP-Adresse bei einer Fertigungssimulation im Netzwerk
-  registrieren, um von dieser Daten zu erhalten.
-\item[FA400] Die Überwachungskonsole kommuniziert über OPC UA in einem TCP/IP Netzwerk mit einer Fertigungssimulation. 
-\item[FA410] Die Überwachungskonsole verbindet sich mit der Fertigungssimulation über deren IP Adresse. 
-\item[FA420] Der Port der Überwachungskonsole kann eingestellt werden.
+\item[FA310] Es gibt eine .jar-Datei, die bei Ausführung die \gls{Uberwachungskonsole} startet.
+\item[FA320] Die \gls{Uberwachungskonsole} enthält einen OPC UA Client.
+\item[FA330] Der OPC UA Client erhält die nicht änderbare IP-Adresse des verwendeten Computers.
```

Comment:
Ich würde den Teil nicht weglassen. Als ich diese Anforderung geschrieben habe, habe ich im Hinterkopf gehabt, dass das Pflichtenheft die Funktionsweise genau spezifiziert und eindeutig festgelegt - damit auch keinen Interpretationsspielraum lässt.
Es mag klar sein, dass die IP-Adresse vom Computer kommt und diese für die Überwachungskonsole nicht änderbar ist (dahin habe ich es umformuliert). Würde es aber nicht da stehen, könnte man das ganze so interpretieren, dass die Überwachungskonsole keine IP-Adresse erhält, irgendeine, diese vom Himmel fällt o. ä. passiert. Außerdem ist dann nicht klar, ob sie geändert werden kann oder nicht. Dann könnte es sein, dass (mal böse ausgedrückt) jemand so schlau (oder auch nicht) ist und die IP-Adresse in den Einstellungen veränderbar macht, was nur graphisch angezeigt wird; intern aber wird diese Adresse nicht verwendet. Damit würde in der Fertigungssimulation die interne Adresse der Überwachungskonsole angezeigt werden, die sich zu der dem Nutzer angezeigten IP-Adresse unterscheidet. Der Nutzer wiederum würde darin einen Fehler sehen, da die angezeigten IP-Adressen unterschiedlich sind. Zusätzlich kann dies zu Divergenzen beim Debuggen führen.
Analog zu den Ports: Wenn die Einschränkung nicht im Pflichtenheft stehen würde, könnte theoretisch die Eingabe eines Ports -1 möglich sein, obwohl das bekanntermaßen nicht möglich sein darf und ist.
Die Analogie mit der Maus, finde ich, passt nicht ganz. Genau genommen schreiben wir schon ins Pflichtenheft, welche Knöpfe es gibt und was bei deren Auslösung passiert; aber nicht, wie sie ausgelöst werden, da dies bereits durch die GUI festgelegt ist.